### ISSUE TITLE
import error: cannot import name 'Protocol' from 'typing'

### ISSUE THREAD

**Issue by hackinteach** [theY4Kman/pytest-drf#5]
Hi, I am trying to import `url_for` into my code and i face the following issue
```
Traceback (most recent call last):
  File "/Users/hackinteachk./.virtualenvs/gfdp/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 180, in console_main
    code = main()
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 136, in main
    config = _prepareconfig(args, plugins)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 314, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/helpconfig.py", line 99, in pytest_cmdline_parse
    config = outcome.get_result()  # type: Config
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 932, in pytest_cmdline_parse
    self.parse(args)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1204, in parse
    self._preparse(args, addopts=addopts)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1097, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 105, in load
    module = import_module(match.group('module'))
  File "/Users/hackinteachk./.pyenv/versions/3.7.3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pytest_drf/__init__.py", line 10, in <module>
    from .views import *
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pytest_drf/views.py", line 18, in <module>
    from pytest_drf.util import deprioritize_base
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pytest_drf/util/__init__.py", line 1, in <module>
    from .expressions import *
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/hackinteachk./.virtualenvs/gfdp/lib/python3.7/site-packages/pytest_drf/util/expressions.py", line 1, in <module>
    from typing import Any, Callable, Iterable, List, Protocol, TypeVar, Union
ImportError: cannot import name 'Protocol' from 'typing' (/Users/hackinteachk./.pyenv/versions/3.7.3/lib/python3.7/typing.py)
```

**Comment by BobMcFry** [comment]
Experienced the same issue yesterday. `Protocol` has been introduced in Python 3.8 and therefore breaks for everyone using lower Python versions.

**Comment by theY4Kman** [comment]
Oh, I fixed this yesterday, but got distracted and forgot to publish. Gimme a few minutes, I'll have this sorted.

The fix I drummed up was installing the [typing-extensions package](https://pypi.org/project/typing-extensions/) for versions below 3.8, and importing `Protocol` from there. Also, I added tox to test on python 3.6, 3.7, and 3.8, to ensure this kind of incompatibility doesn't arise again.
